packages/blade/src/components/BottomSheet/_decisions/decisions.md

```diff
@@ -0,0 +1,245 @@
+
+| Prop       | Type              | Default     | Description                 | Required |
+| ---------- | ----------------- | ----------- | --------------------------- | -------- |
+| `children` | `React.ReactNode` | `undefined` | Contents of the BottomSheet | ✅       |
+
+### `BottomSheetHeader` API
+
+| Prop     | Type            | Default     | Description                          | Required |
+| -------- | --------------- | ----------- | ------------------------------------ | -------- |
+| `title`  | `string`        | `undefined` | Title of the Header                    | ✅       |
+| `leading` | `IconComponent` | `undefined` | leading icon placed before title text |          |
+
+### `BottomSheetFooter` API
+
+| Prop     | Type            | Default     | Description                          | Required |
+| -------- | --------------- | ----------- | ------------------------------------ | -------- |
+| `title`  | `string`        | `undefined` | Title of the BottomSheet footer      | ✅       |
+| `leading` | `IconComponent` | `undefined` | leading icon placed before title text |          |
+| `trailing` | `React.ReactNode` | `undefined` | Renders trailing content |          |
+
+## Composition with Dropdown
+
+We will export `BottomSheet` component separately as an independant component but generally the pattern will be to use it with SelectInput, where in mobile devices the Select's dropdown will be replaced by the BottomSheet.
+
+There are two approaches to doing it:
+
+#### 1. We coupled the BottomSheet & SelectInput tightly and internally conditionally switch the components  
+
+Pros:
+
+- Easy implementation for user's end
+- User's don't have to think about breakpoints, conditional rendering etc
+- From blade side, we will have greater control over the pattern
+
+Cons:
+
+- Bundle size will be impacted, even if users are desktop they will get the bundle of BottomSheet (vice versa)
+- BottomSheet's state will get coupled with SelectDropdown
+- There will be a lot of interdependency of state management
+- These [usecases](https://razorpay.slack.com/archives/C01CS8YBEQZ/p1677825334856589?thread_ts=1677825092.305089&cid=C01CS8YBEQZ) will be harder to solve for without the BottomSheet being an independent component.
+
+
+#### 2. We expose BottomSheet as an independently component and let user do the composition
+
+
+Pros:
+
+- No uneccesary bundle size impact for any of the platforms
+- BottomSheet is decoupled from the state and internal logic of SelectDropdown & can also be used independently
+- Reduced implementation complexity & [interdependency](https://www.notion.so/BottomSheet-API-Discussion-e2aa79cd45274ef280fdb998efa6b98f?pvs=4#630d5e93897b4ca0bf54f475f0cd7d6e) of state
+
+Cons:
+
+- Not trivial to implement from user's end, they will have to compose the BottomSheet & Select as per their needs.
+
+
+Considering the bundle size downside to approach 1, we decided to go ahead with approach 2.
+
+
+> **API Rabbit Hole**  
```

Comment:
can you just copy paste the contents in this doc itself maybe at the bottom so we don't have to move back and forth between GH and Notion

Reply:
:( just waaaaay to much content to copy paste and put it here, bohot zayada indirection ho jayga. So just put only few important final decisions.

Reply:
Okay agree on that, maybe better to keep everything on our end only. I will copy paste the content into a different md file and link it.